demos/python_demos/image_inpainting_demo/image_inpainting_demo.py

```diff
@@ -33,30 +34,103 @@ def build_argparser():
+    args.add_argument("--no_show", help="Optional. Don't show output. Skipped in GUI mode", action='store_true')
+    args.add_argument("-o", "--output", help="Optional. Save output to the file with provided filename. Skipped in GUI mode",
+                      default="", type=str)
+    args.add_argument("-a", "--auto", help="Optional. Use automatic (non-interactive) mode instead of GUI", action='store_true')
 
     return parser
 
-def main():
-    args = build_argparser().parse_args()
+
+def loadImage(imgPath):
+    img = cv2.imread(imgPath, cv2.IMREAD_COLOR)
+    if img is None:
+        print(f"Cannot load image {imgPath}")
+        exit()
+    return img
+
+
+def createRandomMask(parts, max_vertex, max_length, max_brush_width, h, w, max_angle=360):
+    mask = np.zeros((h,w,1), dtype=np.float32)
+    for _ in range(parts):
+        num_strokes = np.random.randint(max_vertex)
+        start_y = np.random.randint(h)
+        start_x = np.random.randint(w)
+        for i in range(num_strokes):
+            angle = np.random.random() * np.deg2rad(max_angle)
+            if i % 2 == 0:
+                angle = 2 * np.pi - angle
+            length = np.random.randint(max_length + 1)
+            brush_width = np.random.randint(10, max_brush_width + 1) // 2 * 2
+            next_y = start_y + length * np.cos(angle)
+            next_x = start_x + length * np.sin(angle)
+
+            next_y = np.clip(next_y, 0, h - 1).astype(np.int)
+            next_x = np.clip(next_x, 0, w - 1).astype(np.int)
+            cv2.line(mask, (start_y, start_x), (next_y, next_x), 1, brush_width)
+            cv2.circle(mask, (start_y, start_x), brush_width // 2, 1)
+
+            start_y, start_x = next_y, next_x
+    return mask
+
+
+def inpaintRandomHoles(args):
+    if len(args.mask_color) != 3:
+        print("Invalid mask_color is provided. Please provide 3 RGB components\n")
+        exit()
+    mask_color=args.mask_color[::-1] # argument comes in RGB mode, but we will use BGR notation below
 
     ie = IECore()
-    inpainting_processor = ImageInpainting(ie, args.model, args.parts,
-                                           args.max_brush_width, args.max_length, args.max_vertex, args.device)
+    img = loadImage(args.input)
+
+    inpainting_processor = ImageInpainting(ie, args.model, args.device)
+
+    #--- Resize to model input and generate random mask
+    img = cv2.resize(img, (inpainting_processor.input_width,inpainting_processor.input_height))
 
-    img = cv2.imread(args.input, cv2.IMREAD_COLOR)
-    masked_image, output_image = inpainting_processor.process(img)
+    if args.rnd:
+        mask = createRandomMask(args.parts,args.max_vertex,args.max_length,args.max_brush_width,
+                                inpainting_processor.input_height,inpainting_processor.input_width)
+    else:
+        top = np.full(img.shape,mask_color,np.uint8)
+        mask = cv2.inRange(img,top,top) / 255
```

Comment:
Nope. It will compare RGB components separately, but we need comparison for all 3 components in one place. It can be done using additional boolean operations, but it would be similar to existing code by complexity

Reply:
Fair enough.